src/app/get-form-overview/getFormOverviewRequestHandler.ts

```diff
@@ -1,85 +1,141 @@
+    const databaseResult = await this.database.getSubmissionsByFormName({ formName: params.formuliernaam });
+    // TODO: empty result different return
+    if (!databaseResult || !Array.isArray(databaseResult)) {
+      throw Error('Cannot retrieve formOverview. DatabaseResult is false or not the expected array.');
+    }
+    // No results from database should return an empty object to throw a 204
+    if (!databaseResult.length) { return { submissions: [], formdefinition: '' };}
 
+    const submissions: string[] = databaseResult.map((dbItem) => {
+      return `${dbItem.key}/submission.json`;
+    });
+    const formdefinition: string =`${databaseResult[0].key}/formdefinition.json`;
+    return { submissions, formdefinition };
   }
 
   async getSubmissionsFromKeys(allKeys: string[]): Promise<GetObjectCommandOutput[]> {
     return this.storage.getBatch(allKeys);
   }
 
-  async compileCsvFile(bucketObjects: GetObjectCommandOutput[]): Promise<string> {
+  async compileCsvFile(bucketObjects: GetObjectCommandOutput[], formParser: FormParser): Promise<string> {
     const csvArray = [];
-    const csvHeaders = ['Formuliernaam', 'DatumTijdOntvangen', 'Kenmerk'];
-    csvArray.push(csvHeaders);
+    const headers = formParser.getHeaders();
+    csvArray.push(headers);
+
     const failedCsvProcessing = [];
+    let headerAndFieldMismatches = 0;
     for (const bucketObject of bucketObjects) {
       if (bucketObject.Body) {
         const bodyString = await bucketObject.Body.transformToString();
-        const jsonData = JSON.parse(bodyString);
-        const jsonMessage = JSON.parse(jsonData.Message);
-        const csvData = [
-          jsonMessage.formTypeId,
-          jsonData.Timestamp,
-          jsonMessage.reference,
-        ];
-        csvArray.push(csvData);
-
+        const parsedForm = formParser.parseForm(bodyString);
+        if (parsedForm.length !== headers.length) { headerAndFieldMismatches++;}
+        csvArray.push(parsedForm);
       } else {
-        failedCsvProcessing.push(`Geen body. Mogelijk metadata requestId: ${bucketObject.$metadata.requestId}`);
+        failedCsvProcessing.push(`No form body retrieved from body. Possibly only metadata retrieved with requestId: ${bucketObject.$metadata.requestId}`);
       }
     }
-    let csvContent: string = '';
 
+    let csvContent: string = '';
     csvArray.forEach(row => {
       csvContent += row.join(',') + '\n';
     });
-    console.log('Aantal verwerkte csv rijen: ', (csvArray.length - 1));
-    console.log('Failed csv transformations. Count: ', failedCsvProcessing.length, ' Objects failed: ', failedCsvProcessing);
-
+    console.log(`Done processing csv file. Number of processed rows: ${(csvArray.length - 1)}. Number of failed csv transformations: ${failedCsvProcessing.length}. Number of header and form fields length mismatches:  ${headerAndFieldMismatches}.`);
```

Comment:
Deze logging is gemakkelijk te negeren. Moeten we bij mismatches een error loggen? Of evt. deze statusinfo in de csv opnemen (wat suboptimaal is qua verwerking)?

Reply:
Uiteindelijk wel, maar er zijn momenteel een aantal formulieren die nog niet optimaal verwerkt kunnen worden.
Ik wil errors voorkomen.